compare.py

```diff
@@ -0,0 +1,314 @@
+                    "status": self.status[-3],
+                    "nvr1": None,
+                    "nvr2": build["nvr"],
+                }
+
+    def compare_all(self):
+        for package in content:
+            logging.debug(f'Processing package {package}')
+            self.results[package] = self.compare_one(package)
+        return self.results
+
+    def count(self):
+        stats = {}
+        for item in self.results.values():
+            value = item["status"]
+            if value not in stats:
+                stats[value] = 0
+            stats[value] += 1
+        stats["total"] = sum(stats.values())
+        return stats
+
+    def render(self, tmpl_path="templates", output_path="output", fmt="all"):
+        os.makedirs(output_path, exist_ok=True)
+
+        j2_env = jinja2.Environment(loader=jinja2.FileSystemLoader(tmpl_path))
+        templates = j2_env.list_templates(extensions="j2")
+        if fmt != "all":
+            fmtlist = fmt.split(",")
+            templates = [
+                name for name in templates if name.split(".")[-2] in fmtlist
+            ]
+
+        for tmpl_name in templates:
+            tmpl = j2_env.get_template(tmpl_name)
+            tmpl.stream(
+                source1=self.source1,
+                source2=self.source2,
+                results=self.results,
+                stats=self.count(),
+                date=datetime.datetime.now()
+            ).dump(
+                os.path.join(
+                    output_path,
+                    tmpl_name[:-3],
+                )
+            )
+
+
+def get_content(distro_view="eln"):
+    """Builds the full list of packages for the distro from the Content Resolver
+
+    Merges result for all architectures.
+    """
+    merged_packages = set()
+
+    distro_url = "https://tiny.distro.builders"
+    arches = ["aarch64", "armv7hl", "ppc64le", "s390x", "x86_64"]
+    which_source = ["source", "buildroot-source"]
+
+    for arch in arches:
+        for this_source in which_source:
+            url = (
+                "{distro_url}"
+                "/view-{this_source}-package-name-list--view-{distro_view}--{arch}.txt"
+            ).format(distro_url=distro_url, this_source=this_source, distro_view=distro_view, arch=arch)
+
+            logging.debug("downloading {url}".format(url=url))
+
+            r = requests.get(url, allow_redirects=True)
+            for line in r.text.splitlines():
+                merged_packages.add(line)
+
+    logging.debug("Found a total of {} packages".format(len(merged_packages)))
+
+    return merged_packages
+
+
+def evr(build):
+    """Get epoch, version, release data from the build
+
+    We currently reset Epoch value to 0, because we have number of cases where
+    epoch of a package in Rawhide is different from that of ELN.
+
+    We remove dist tag data from the release, so that we can compare nvr's
+    between different distributions.
+    """
+
+    epoch = "0"
```

Comment:
Is it necessary to create a separate variable here?
Couldn't find where it uses instead of `return` .